client/components/login/login.js

```diff
@@ -206,27 +195,25 @@ export default class Login extends React.Component {
       }),
     })
       .then((res = {}) => {
-        if (res.data) setIsActive(res.data.is_active);
-        return handleAuthentication(
-          settings.mobile_phone_verification,
-          res.data,
-        );
+        return handleAuthentication(res.data);
       })
       .catch((error) => {
         const {data} = error.response;
-        setIsActive(data.is_active);
         if (
           error.response.status === 401 &&
           settings.mobile_phone_verification &&
           data.is_active
         ) {
-          return handleAuthentication(true);
+          return handleAuthentication(data);
         }
+        userData.is_active = false;
+        setUserData(userData);
         const errorText = data.is_active
           ? getErrorText(error, loginError)
           : getErrorText(error, userInactiveError);
         logError(error, errorText);
         toast.error(errorText);
+        if (data.is_active === false) data.username = "";
```

Comment:
why is this needed? What problem does it solve?
If the user is inactive, won't they just be logged out? Then why would we need to set the username as empty string?